lib/spack/spack/spec.py

```diff
@@ -773,18 +796,37 @@ def constrain(self, other):
             elif k not in self:
                 self[k] = other[k]
                 changed = True
+            # Check that propagation
```

Comment:
Did you want to complete this comment (i.e., `Check that propagation <something>`)?